### PR TITLE
Enable signing testFixtureJar for maven release

### PR DESCRIPTION
**Pull request by iamgusain**
Enabling the support for signing testfixture jar when creating maven package for publishing.
Currently it's not signed, and it fails verification when releasing to external maven with error "Missing Signature" for the testfixture jar

### CODE REVIEW

File: azure-pipelines/maven-release/common4j.yml [hunk 1]
```diff
@@ -28,3 +28,4 @@ jobs:
     gpgSourcesJar: true
     gpgJavadocJar: true
     gpgJar: true
+    testFixtureJar: true
```
Comment (p3dr0rv) [inline azure-pipelines/maven-release/common4j.yml]:
```suggestion
    gpgTestFixtureJar: true
```

File: azure-pipelines/templates/steps/maven-release/gpg-signing.sh [hunk 2]
```diff
@@ -73,6 +73,18 @@ if [ $JAVADOCJAR == "true" ]; then
     fi
 fi
 
+if [ $TESTFIXUTREJAR == "true" ]; then
```
Comment (p3dr0rv) [inline azure-pipelines/templates/steps/maven-release/gpg-signing.sh]:
```suggestion
if [ $TESTFIXTUREJAR == "true" ]; then
```

**Review by p3dr0rv — CHANGES_REQUESTED** [review]

**Comment by shahzaibj** [comment]
It seems like the bigger issue here is that testFixtures are being released to external maven? Should we try to fix that rather than signing these?

**Comment by iamgusain** [comment]
> It seems like the bigger issue here is that testFixtures are being released to external maven? Should we try to fix that rather than signing these?

testFixture jar is by default generated by the testFixture plugin, in our maven release we are copying all the generated jars as part of the maven package. We can exclude it there but I thought we included these as testFixture so the consumers could use them for testing. If these are internal only, should these be part of our other testUtil libs.

**Comment by shahzaibj** [comment]
> > It seems like the bigger issue here is that testFixtures are being released to external maven? Should we try to fix that rather than signing these?
> 
> testFixture jar is by default generated by the testFixture plugin, in our maven release we are copying all the generated jars as part of the maven package. We can exclude it there but I thought we included these as testFixture so the consumers could use them for testing. If these are internal only, should these be part of our other testUtil libs.

Interesting question. @p3dr0rv Any thoughts here?

**Review by somalaya — APPROVED** [review]

**Pull request closed without merging**